bootstrapper/pkg/clients/instance_config_client.go

```diff
@@ -45,6 +45,41 @@ func (c *InstanceConfigClient) ConfigureInstallationDefaults(params map[string]s
 	return c.configure(params, protoconfig.ConfigType_CONFIG_TYPE_INSTALLATION_DEFAULTS, "Installation defaults")
 }
 
+// GetInstallationID returns the installation ID from the instance configuration.
+// If the configuration is not found or not in configured state, returns an empty string.
+func (c *InstanceConfigClient) GetInstallationID() string {
+	return c.getConfigField("installation_id")
+}
+
+// getConfigField retrieves a specific field from the installation defaults configuration.
+// If the configuration is not found or not in configured state, returns an empty string.
+func (c *InstanceConfigClient) getConfigField(field string) string {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+
+	req := &protoconfig.ListConfigsRequest{
+		Types: []protoconfig.ConfigType{protoconfig.ConfigType_CONFIG_TYPE_INSTALLATION_DEFAULTS},
+	}
+
+	resp, err := c.client.ListConfigs(ctx, req)
+	if err != nil {
+		log.Errorf("Failed to list installation configurations: %v", err)
+		return ""
+	}
+
+	for _, config := range resp.Configs {
+		if config.State == protoconfig.State_STATE_CONFIGURED {
+			for _, f := range config.Fields {
+				if f.Key == field {
+					return f.Value
+				}
+			}
+		}
+	}
+
```

Comment:
WDYT about adding some logging here? It'll be easier to spot configuration problems.